src/movies.js

```diff
@@ -1,16 +1,40 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+    const directorsNames = movies.map((movie) => {
+      return movie.director
+    }); 
+      return directorsNames; 
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
-
+function howManyMovies(movies) {
+  const countMovies = movies.filter((movie) => {
+    if (movie.director === "Steven Spielberg" && movie.genre.indexOf('Drama') !== -1) {
+      return movie;
+    }    
+  });
+  return countMovies.length;
+}
+    
+ 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  const average = movies.reduce((acc, curr) => {
+    console.log(`${acc} .... ${curr}`);
+  return acc + curr;
```

Comment:
When working with objects, pay attention if you need to access full object or its property. In this case you only want to be adding score.

Movies -> it is array of movie objects
Curr -> is single movie object

So what you want to be adding is: acc + curr.score